project-base/app/src/Model/Product/Product.php

```diff
@@ -335,11 +336,18 @@ public function getCatnum(): string
     }
 
     /**
+     * @param int|null $limit
      * @return \App\Model\Product\Product[]
      */
-    public function getRelatedProducts(): array
+    public function getRelatedProducts(?int $limit = null): array
```

Comment:
I would limit only usages in ProductArrayFieldMapper and ProductEntityFieldMapper. 
The intention seems clearer to me then

Reply:
I am not sure about limiting the collection 🤔 
Is it necessary?